and-cli-deploy-azure-storage.js

```diff
@@ -0,0 +1,193 @@
+#!/usr/bin/env node
+require("./command-runner").run(async () => {
+    // -----------------------------------------------------------------------------------------
+    // #region Imports
+    // -----------------------------------------------------------------------------------------
+
+    const azure          = require("./_modules/azure");
+    const echo           = require("./_modules/echo");
+    const file           = require("./_modules/file");
+    const frontendPath   = require("./_modules/frontend-path");
+    const path           = require("path");
+    const program        = require("commander");
+    const shell          = require("shelljs");
+    const upath          = require("upath");
+    const webpackPublish = require("./_modules/webpack-publish");
+
+    // #endregion Imports
+
+    // -----------------------------------------------------------------------------------------
+    // #region Variables
+    // -----------------------------------------------------------------------------------------
+
+    let   clientId            = null;
+    let   destination         = null;
+    const environmentFileName = ".env.local";
+    const pythonInstallerUrl  = "https://www.python.org/ftp/python/3.7.4/python-3.7.4-amd64.exe";
+    let   recursive           = false;
+    let   secret              = null;
+    let   sourcePath          = frontendPath.publishDir() + "/*";
+    let   tenantId            = null;
+    let   username            = null;
+
+    // #endregion Variables
+
+    /// -----------------------------------------------------------------------------------------
+    // #region Functions
+    // -----------------------------------------------------------------------------------------
+
+    const deployAzureStorage = {
+        cmd() {
+            let command = `az storage copy -s ${sourcePath} -d ${destination}`;
+
+            if (recursive) {
+                command += " --recursive";
+            }
+
+            return command;
+        },
+        description() {
+            return "Publish build artifacts to Azure Storage";
+        },
+        run() {
+            // Check system/command requirements
+            this.validateOrExit();
+
+            // Configure .env.local to set public url before publish
+            if (program.publicUrl) {
```

Comment:
Suggestion: Believe this is the same as the block in the `and-cli-deploy-aws-s3.js` file. Consider breaking into a shared module that is shared between them.

Reply:
Good call!